### DISCUSSION TITLE
.NET 9 RC 2

### DISCUSSION THREAD

**Discussion by jamesmontemagno** [dotnet/core#9534] [News]
We are pleased to announce the release of .NET 9 Release Candidate 2. As we prepare the general availability (GA) release of .NET 9 in November we are adding our final touches around performance, stability, and any additional optimizations to make it the best release of .NET 9.  This release comes with a [go-live license](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#previews), indicating that it is supported in production environments. We encourage developers today to try out this latest release and give the teams feedback on anything that you find in your development so we can incorporate any additional enhancements into the final release.

You can find release notes for enhancements in this release for the following:

* [📚Libraries](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/libraries.md)
* [⏱️Runtime](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/runtime.md)
* [🛠️ SDK](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/sdk.md)
* [🌐 ASP.NET Core](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/aspnetcore.md) with an [update to latest Bootstrap, jQuery, and jQuery Validation versions](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/aspnetcore.md#updated-to-latest-bootstrap-jquery-and-jquery-validation-versions)
* [📱 .NET MAUI](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/dotnetmaui.md) including [Xcode 16 and latest platform SDK versions](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/dotnetmaui.md#xcode-16-and-latest-platform-sdk-versions)

## Get started

To get started with .NET 9, [install the .NET 9 SDK](https://dotnet.microsoft.com/download/dotnet/9.0).

If you're on Windows using Visual Studio, we recommend installing the latest [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs/preview/). .NET 9 is now can be installed directly through the Visual Studio installer starting with Visual Studio 2022 17.12 Preview 3 (Available Oct. 15th).

You can also use Visual Studio Code and the [C# Dev Kit](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit) extension with .NET 9.

You can find [.NET 9 RC 2 Installer & Binaries](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/rc2/9.0.0-rc.2.md) on GitHub.

## Stay up-to-date with .NET 9

You can stay up-to-date with all the features of .NET 9 with:

* [What's new in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview)
* [What's new in ASP.NET Core](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-9.0)
* [What's new in .NET MAUI](https://learn.microsoft.com/dotnet/maui/whats-new/dotnet-9)
* [What's new in EF Core](https://learn.microsoft.com/ef/core/what-is-new/ef-core-9.0/whatsnew)
* [Breaking Changes in .NET 9](https://learn.microsoft.com/dotnet/core/compatibility/9.0)
* [.NET 9 Releases](https://github.com/dotnet/core/blob/main/release-notes/9.0/README.md)

Be sure to subscribe to the GitHub Discussions [news feed](https://github.com/dotnet/core/discussions/categories/news.atom) for all release announcements.

**Comment by GerardSmit** [comment]
The release notes for Libraries, Runtime and SDK are empty. For people that want to check what has been fixed, here are the PRs that were merged (excluding dependency merges):
- [Libraries / Runtime](https://github.com/dotnet/runtime/pulls?q=is%3Apr+base%3Arelease%2F9.0+merged%3A2024-09-09..2024-10-09+NOT+%22Update+dependencies+from%22)
- [SDK](https://github.com/dotnet/sdk/pulls?q=is%3Apr+base%3Arelease%2F9.0+merged%3A2024-09-09..2024-10-09+NOT+%22Update+dependencies+from%22)
- [ASP.NET Core](https://github.com/dotnet/aspnetcore/pulls?q=sort%3Aupdated-desc+is%3Apr+base%3Arelease%2F9.0+merged%3A2024-09-09..2024-10-09+NOT+%22Update+dependencies+from%22+NOT+%22%28deps%29%22)
- [.NET MAUI](https://github.com/dotnet/maui/pulls?q=is%3Apr+base%3Arelease%2F9.0+merged%3A2024-09-09..2024-10-09+NOT+%22Update+dependencies+from%22)

<details>
  <summary><i>Outdated comments</i></summary>
  
Also, some broken links that I found:

1. https://github.com/dotnet/core/releases/tag/v9.0.0-rc.2: "Release Notes" links to the directory "rc1", so it results in a 404.
2. https://get.dot.net/9: Announcement (alert on top of the page) shows ".NET 9.0.0-rc.2 is now available." but links to RC1.
  
</details>

**Reply by jamesmontemagno** [reply]
Thanks, yes there were no new features added which is that we highlight in the release notes. We opted to go for a general message in all of them. I have updated them to have a changelog notice release to release tag, which I think is most representative, but will work with the teams. https://github.com/dotnet/core/pull/9538

**Reply by mairaw** [reply]
And thanks for reporting the broken links. Our team has fixed them.

**Comment by pjmlp** [comment]
I keep wondering if .NET 9 will bring any updates to F#, given its continuous absence from  anything .NET 9 related.

**Reply by baronfel** [reply]
It definitely will - you can check out this video I did last month with Vlad from the F# team where we talk about some of the awesome features coming down the pipe: https://www.youtube.com/watch?v=BrI4jeq7fJM

In addition, the F# team is _incredibly_ good about documenting changes and features coming in their release notes. You can find release notes for .NET 9.0.100 for the [F# Compiler Services](https://github.com/dotnet/fsharp/blob/main/docs/release-notes/.FSharp.Compiler.Service/9.0.100.md), [FSharp.Core](https://github.com/dotnet/fsharp/blob/main/docs/release-notes/.FSharp.Core/9.0.100.md) library, the [F#](https://github.com/dotnet/fsharp/blob/main/docs/release-notes/.Language/9.0.md) language itself, and the [Visual Studio Integration](https://github.com/dotnet/fsharp/blob/main/docs/release-notes/.VisualStudio/17.12.md) all in the repo directly.

**Reply by pjmlp** [reply]
Thanks for the video, still I would like to point out having the F# updates not showing up alongside anything .NET 9 related feels kind of awkward. 

We start to wonder if it is going down the same path as Visual Basic and C++/CLI, given the usual Microsoft behaviour to go silent.

**Comment by malsabi** [comment]


**Reply by mairaw** [reply]
@malsabi can you clarify your question? .NET 8 will still be available and supported until 2026.

**Comment by pedroreys** [comment]
fyi, .NET 9 RC2 is not listed in the [go-live license](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#previews) page that y'all linked. Only RC1 is listed there.

![_NET_and__NET_Core_official_support_policy](https://github.com/user-attachments/assets/e8487743-3c3f-4d09-8e8c-5cc1c07755d2)

**Reply by mairaw** [reply]
~Yes, we're publishing the updated page with that information.~

The page has been updated.

**Comment by fmaeseele** [comment]
".NET 9 is now can be installed directly through the Visual Studio installer starting with Visual Studio 2022 17.12 Preview 3."

Where is Visual Studio 2022 17.12 Preview 3 ? Only Preview 2.1 shows up.

**Reply by jamesmontemagno** [reply]
Yes, sorry looks like a slight delay and should be out next week. Will update here.

**Reply by dlemstra** [reply]
Would it be possible to also update this post with the date: https://devblogs.microsoft.com/dotnet/dotnet-9-rc-2/? Took me a while before I figured out why Preview 3 was not yet available.

**Reply by MagicAndre1981** [reply]
> Took me a while before I figured out why Preview 3 was not yet available.

me, too. Just found this discussion to get the info that 17.12 preview 3 is delayed.

**Reply by mairaw** [reply]
> Would it be possible to also update this post with the date: https://devblogs.microsoft.com/dotnet/dotnet-9-rc-2/? Took me a while before I figured out why Preview 3 was not yet available.

Updated the blog too.

**Comment by DavidBrower** [comment]
Is there a definitive date for the GA release?

**Reply by MagicAndre1981** [reply]
I think tomorrow as .[net conf](https://www.dotnetconf.net/) starts here and day 1 is about .net9:

> Day one is all about the big news, .NET 9! Join the .NET team on all the new things you can do with the latest release.

**Reply by WeihanLi** [reply]
According to the previous releases, should be November 12, 2024, will be released before the dotnetconf starts

**Reply by martincostello** [reply]
November 12th also aligns exactly with "Patch Tuesday" for November (second Tuesday of the month), which is typically when .NET releases are published.
